src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -87,20 +109,15 @@ ShardManager::Result< ShardInfo > HSHomeObject::_create_shard(pg_id pg_owner, ui
     raw_ptr += sizeof(ReplicationMessageHeader);
     std::memcpy(raw_ptr, create_shard_message.c_str(), create_shard_message.size());
 
-    sisl::sg_list value;
     // replicate this create shard message to PG members;
-    auto [p, sf] = folly::makePromiseContract< ShardManager::Result< ShardInfo > >();
-    replica_set->write(sisl::blob(buf->data_begin(), needed_size), sisl::blob(), value, static_cast< void* >(&p));
-    auto info = std::move(sf).get();
-    if (!bool(info)) {
-        LOGWARN("create new shard [{}] on pg [{}] is failed with error:{}", new_shard_id & shard_mask, pg_owner,
-                info.error());
-    }
+    pg->repl_dev_->async_alloc_write(req->hdr_buf_, sisl::blob{}, sisl::sg_list{}, req);
```

Comment:
The PGID/ShardID needed to be included into header. 
https://github.com/eBay/HomeObject/pull/68/files will do that.